game/CMakeLists.txt

```diff
@@ -108,55 +108,90 @@ foreach(lib ${OUR_LIBS} SDL2 PangoCairo LibRSVG LibXML++ AVFormat SWResample SWS
 endif()
 
-option(ENABLE_WEBCAM "Enable webcam support (OpenCV)." ON)
-if(ENABLE_WEBCAM)
-	find_package(OpenCV)
+if(PortMidi_FOUND)
+	include_directories(${PortMidi_INCLUDE_DIRS})
+	list(APPEND LIBS ${PortMidi_LIBRARIES})
+	add_definitions("-DUSE_PORTMIDI")
+endif()
+
+# Activating webcam
+set(ENABLE_WEBCAM AUTO CACHE STRING "Use webcam")
+set_property(CACHE ENABLE_WEBCAM PROPERTY STRINGS AUTO ON OFF)
+if(NOT DEFINED ENABLE_WEBCAM OR "AUTO" STREQUAL ${ENABLE_WEBCAM})
+	# webcam support not requested explicitly, best effort try to find it
+	find_package(OpenCV QUIET)
 	if(OpenCV_FOUND)
-		include_directories(${OpenCV_INCLUDE_DIRS})
-		list(APPEND LIBS ${OpenCV_LIBS})
-		add_definitions("-DUSE_OPENCV")
-		message(STATUS "Webcam support: Enabled")
+		message(STATUS "Webcam support: Enabled (automatically found)")
 	else()
-		message(STATUS "Webcam support: Disabled (libcv/libhighgui not found)")
+		message(STATUS "Webcam support: Disabled (OpenCV (libcv/libhighgui) not found)")
 	endif()
+elseif(ENABLE_WEBCAM)
+	# webcam support explicitly requested, make it mandatory
+	find_package(OpenCV REQUIRED)
+	message(STATUS "Webcam support: Enabled (explicitly enabled)")
 else()
+	# webcam support explicitly disabled
 	message(STATUS "Webcam support: Disabled (explicitly disabled)")
 endif()
 
-option(ENABLE_WEBSERVER "Enable webserver support (cpprestsdk)." ON)
-if(ENABLE_WEBSERVER)
-	find_package(CppRest)
+if(OpenCV_FOUND)
+	include_directories(${OpenCV_INCLUDE_DIRS})
+	list(APPEND LIBS ${OpenCV_LIBS})
+	add_definitions("-DUSE_OPENCV")
+endif()
+
+# Activating webserver
+set(ENABLE_WEBSERVER AUTO CACHE STRING "Use webserver")
+set_property(CACHE ENABLE_WEBSERVER PROPERTY STRINGS AUTO ON OFF)
+if(NOT DEFINED ENABLE_WEBSERVER OR "AUTO" STREQUAL ${ENABLE_WEBSERVER})
+	# webserver not requested explicitly, best effort try to find it
+	find_package(CppRest QUIET)
 	if(CppRest_FOUND)
-		find_package(Boost 1.55 REQUIRED COMPONENTS chrono thread)
-		include_directories(${Boost_INCLUDE_DIRS})
-		list(APPEND LIBS ${Boost_LIBRARIES})
-
-		include_directories(${CppRest_INCLUDE_DIRS})
-		list(APPEND LIBS ${CppRest_LIBRARIES})
-		message(STATUS "Webserver support: Enabled")
-		add_definitions("-DUSE_WEBSERVER")
+		message(STATUS "Webserver support: Enabled (automatically found)")
 	else()
 		message(STATUS "Webserver support: Disabled (cpprestsdk not found)")
 	endif()
+elseif(ENABLE_WEBSERVER)
+	# webserver explicitly requested, make it mandatory
+	find_package(CppRest REQUIRED)
+	message(STATUS "Webserver support: Enabled (explicitly enabled)")
 else()
+	# webserver explicitly disabled
 	message(STATUS "Webserver support: Disabled (explicitly disabled)")
 endif()
 
+# if cpprest is found, add it to the dependencies
+if(CppRest_FOUND)
+	include_directories(${CppRest_INCLUDE_DIRS})
+	list(APPEND LIBS ${CppRest_LIBRARIES})
+	add_definitions("-DUSE_WEBSERVER")
+
+	# add other required dependencies then
+	find_package(Boost 1.55 REQUIRED COMPONENTS chrono thread)
```

Comment:
this would really ned to be remove from here, but the code was already like that... just a head up

Reply:
I've added a FIXME

Reply:
thank you